server/src/main/java/org/opensearch/search/backpressure/SearchBackpressureService.java

```diff
@@ -222,13 +259,19 @@ TaskCancellation getTaskCancellation(CancellableTask task) {
         for (TaskResourceUsageTracker tracker : taskResourceUsageTrackers) {
             Optional<TaskCancellation.Reason> reason = tracker.checkAndMaybeGetCancellationReason(task);
             if (reason.isPresent()) {
+                if (task instanceof SearchTask) {
```

Comment:
Good point, I have introduced a new set of trackers for SearchTasks. This will also be useful if we want to have different set of trackers for different types of tasks.